controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -1122,6 +1124,177 @@ bool master_thread::handle_cmdu_1905_channel_selection_response(Socket *sd,
     return true;
 }
 
+static void print_link_metric_map(
+    std::map<sMacAddr, std::map<sMacAddr, son::node::link_metrics_data>> &m_link_metric_data)
+{
+    LOG(DEBUG) << "Printing Link Metrics data map";
+    for (auto const &pair_agent : m_link_metric_data) {
+        LOG(DEBUG) << "sent from al_mac= " << network_utils::mac_to_string(pair_agent.first)
+                   << std::endl;
+
+        for (auto const &pair_neighbor : m_link_metric_data[pair_agent.first]) {
+            LOG(DEBUG) << "reporting neighbor al_mac= "
+                       << network_utils::mac_to_string(pair_neighbor.first) << std::endl;
+
+            auto &vrx = pair_neighbor.second.receiverLinkMetrics;
+            for (unsigned int i = 0; i < vrx.size(); ++i) {
+                LOG(DEBUG)
+                    << "rx interface metric data # " << i << "  neighbor al_mac:"
+                    << network_utils::mac_to_string(vrx[i].mac_of_an_interface_in_the_neighbor_al)
+                    << "  receiving al_mac:"
+                    << network_utils::mac_to_string(vrx[i].mac_of_an_interface_in_the_receiving_al)
+                    << "  rssi= " << std::hex << int(vrx[i].link_metric_info.rssi_db) << std::endl;
+            }
+
+            auto &vtx = pair_neighbor.second.transmitterLinkMetrics;
+            for (unsigned int i = 0; i < vtx.size(); i++) {
+                LOG(DEBUG)
+                    << "tx interface metric data # " << i << "  neighbor al_mac:"
+                    << network_utils::mac_to_string(vtx[i].mac_of_an_interface_in_the_neighbor_al)
+                    << "  receiving al_mac:"
+                    << network_utils::mac_to_string(vtx[i].mac_of_an_interface_in_the_receiving_al)
+                    << "  phy_rate= " << std::hex << int(vtx[i].link_metric_info.phy_rate)
+                    << std::endl;
+            }
+        }
+    }
+}
+
+bool master_thread::handle_cmdu_1905_link_metric_response(Socket *sd,
+                                                          ieee1905_1::CmduMessageRx &cmdu_rx)
+{
+
+    auto mid = cmdu_rx.getMessageId();
+    LOG(INFO) << "Received LINK_METRIC_RESPONSE_MESSAGE, mid=" << std::dec << int(mid);
+
+    //getting reference for link metric data storage from db
+    auto &m_link_metric_data = database.get_link_metric_data_map(database.get_gw_mac());
+
+    //will hold new link metric data from Reporting Agent
+    son::node::link_metrics_data new_link_metric_data;
+
+    int tlvType;
+    // holds mac address of the reporting agent, used as m_link_metric_data key
+    sMacAddr reporting_agent_al_mac, neighbor_al_mac;
+    reporting_agent_al_mac.struct_init();
+    neighbor_al_mac.struct_init();
```

Comment:
There is no need to call struct_init(). It is meant for use in the tlv init function.